File: scripts/resize-icon.py
#!/usr/bin/env python3
"""
Resize the MeraProduct icon to all required sizes
Usage: python3 scripts/resize-icon.py <path-to-icon>
"""

import sys
import os
from PIL import Image

SIZES = [16, 32, 48, 128]
OUTPUT_DIRS = [
    'assets/icons',
    'dist/assets/icons'
]

def resize_icon(input_path):
    print(f'🖼️  Loading icon from: {input_path}')
    
    if not os.path.exists(input_path):
        print(f'❌ Error: Icon file not found: {input_path}')
        sys.exit(1)
    
    try:
        # Load the original image
        with Image.open(input_path) as img:
            print(f'✅ Original image loaded: {img.size[0]}x{img.size[1]}')
            
            # Convert to RGBA if necessary
            if img.mode != 'RGBA':
                img = img.convert('RGBA')
            
            # Create output directories if they don't exist
            for dir_path in OUTPUT_DIRS:
                os.makedirs(dir_path, exist_ok=True)
                print(f'📁 Created/verified directory: {dir_path}')
            
            # Resize and save for each size
            for size in SIZES:
                print(f'\n🔄 Creating {size}x{size} icon...')
                
                # Resize using high-quality LANCZOS resampling
                resized = img.resize((size, size), Image.Resampling.LANCZOS)
                
                # Save to both directories
                for dir_path in OUTPUT_DIRS:
                    output_path = os.path.join(dir_path, f'icon{size}.png')
                    resized.save(output_path, 'PNG', optimize=True)
                    print(f'   ✅ Saved: {output_path}')
            
            print('\n🎉 All icons created successfully!')
            print('\n📦 Icon locations:')
            for dir_path in OUTPUT_DIRS:
                print(f'   - {dir_path}/')
                for size in SIZES:
                    print(f'     • icon{size}.png')
    
    except Exception as e:
        print(f'❌ Error processing icon: {e}')
        sys.exit(1)

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print('❌ Usage: python3 scripts/resize-icon.py <path-to-icon>')
        print('   Example: python3 scripts/resize-icon.py ~/Downloads/meraproduct-icon.png')
        sys.exit(1)
    
    input_path = os.path.abspath(sys.argv[1])
    resize_icon(input_path)

File: scripts/resize-icon.js
/**
 * Resize the MeraProduct icon to all required sizes
 * Usage: node scripts/resize-icon.js <path-to-icon>
 */

const fs = require('fs');
const path = require('path');
const { createCanvas, loadImage } = require('canvas');

const SIZES = [16, 32, 48, 128];
const OUTPUT_DIRS = [
  path.join(__dirname, '../assets/icons'),
  path.join(__dirname, '../dist/assets/icons')
];

async function resizeIcon(inputPath) {
  console.log('🖼️  Loading icon from:', inputPath);
  
  if (!fs.existsSync(inputPath)) {
    console.error('❌ Error: Icon file not found:', inputPath);
    process.exit(1);
  }

  try {
    // Load the original image
    const image = await loadImage(inputPath);
    console.log(`✅ Original image loaded: ${image.width}x${image.height}`);

    // Create output directories if they don't exist
    OUTPUT_DIRS.forEach(dir => {
      if (!fs.existsSync(dir)) {
        fs.mkdirSync(dir, { recursive: true });
        console.log('📁 Created directory:', dir);
      }
    });

    // Resize and save for each size
    for (const size of SIZES) {
      console.log(`\n🔄 Creating ${size}x${size} icon...`);
      
      // Create canvas
      const canvas = createCanvas(size, size);
      const ctx = canvas.getContext('2d');

      // Enable high-quality image smoothing
      ctx.imageSmoothingEnabled = true;
      ctx.imageSmoothingQuality = 'high';

      // Draw the resized image
      ctx.drawImage(image, 0, 0, size, size);

      // Save to both directories
      const buffer = canvas.toBuffer('image/png');
      
      for (const dir of OUTPUT_DIRS) {
        const outputPath = path.join(dir, `icon${size}.png`);
        fs.writeFileSync(outputPath, buffer);
        console.log(`   ✅ Saved: ${outputPath}`);
      }
    }

    console.log('\n🎉 All icons created successfully!');
    console.log('\n📦 Icon locations:');
    OUTPUT_DIRS.forEach(dir => {
      console.log(`   - ${dir}/`);
      SIZES.forEach(size => console.log(`     • icon${size}.png`));
    });

  } catch (error) {
    console.error('❌ Error processing icon:', error.message);
    process.exit(1);
  }
}

// Get input path from command line
const inputPath = process.argv[2];

if (!inputPath) {
  console.error('❌ Usage: node scripts/resize-icon.js <path-to-icon>');
  console.error('   Example: node scripts/resize-icon.js ~/Downloads/meraproduct-icon.png');
  process.exit(1);
}

resizeIcon(path.resolve(inputPath));
